velox/functions/sparksql/DecimalUtil.h

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <boost/multiprecision/cpp_int.hpp>
+
+#include "velox/type/DecimalUtil.h"
+#include "velox/type/Type.h"
+
+namespace facebook::velox::functions::sparksql {
+using uint128_t = __uint128_t;
+using int256_t = boost::multiprecision::int256_t;
+
+class DecimalUtil {
+ public:
+  template <
+      class T,
+      typename = std::enable_if_t<
+          std::is_same_v<T, int64_t> || std::is_same_v<T, int128_t>>>
+  inline static T convert(int256_t in, bool& overflow) {
+    typedef typename std::
```

Comment:
You can declare this at the class level and use it for all 3 functions below.

Reply:
Because it also use this function, https://github.com/facebookincubator/velox/pull/4613/files#diff-4baee9f82f9c347f753972415d345941d2c2a110b608d480b090650171da096bR259, so I declare it as public